diff --git a/lib/config/config-loader.js b/lib/config/config-loader.js
index 85944672a..2823eeb8a 100644
--- a/lib/config/config-loader.js
+++ b/lib/config/config-loader.js
@@ -39,2 +39,10 @@ const { FlatConfigArray } = require("./flat-config-array");
 
+const DOTCONFIG_FILENAMES = [
+    ".config/eslint.js",
+    ".config/eslint.mjs",
+    ".config/eslint.cjs",
+    ".config/eslint.ts",
+    ".config/eslint.mts",
+    ".config/eslint.cts"
+];
 const FLAT_CONFIG_FILENAMES = [
@@ -460,3 +468,8 @@ class ConfigLoader {
             if (configFilePath) {
                 basePath = path.dirname(configFilePath);
+            } else {
+                configFilePath = await findUp(DOTCONFIG_FILENAMES, { cwd: fromDirectory });
+                if (configFilePath) {
+                    basePath = path.dirname(path.dirname(configFilePath));
+                }
             }
